PhysicsTools/PatAlgos/plugins/PATElectronSlimmer.cc

```diff
@@ -114,10 +127,34 @@ pat::PATElectronSlimmer::produce(edm::Event & iEvent, const edm::EventSetup & iS
     if( modifyElectron_ ) { electronModifier_->setEvent(iEvent); }
     if( modifyElectron_ ) electronModifier_->setEventContent(iSetup);
 
+
+    //value maps for puppi isolation
+    Handle<edm::ValueMap<float>> PUPPIIsolation_charged_hadrons;
+    Handle<edm::ValueMap<float>> PUPPIIsolation_neutral_hadrons;
+    Handle<edm::ValueMap<float>> PUPPIIsolation_photons;
+
+    iEvent.getByToken(PUPPIIsolation_charged_hadrons_, PUPPIIsolation_charged_hadrons);
+    iEvent.getByToken(PUPPIIsolation_neutral_hadrons_, PUPPIIsolation_neutral_hadrons);
+    iEvent.getByToken(PUPPIIsolation_photons_, PUPPIIsolation_photons);
+
+    //value maps for puppiNoLeptons isolation
+    Handle<edm::ValueMap<float>> PUPPINoLeptonsIsolation_charged_hadrons;
+    Handle<edm::ValueMap<float>> PUPPINoLeptonsIsolation_neutral_hadrons;
+    Handle<edm::ValueMap<float>> PUPPINoLeptonsIsolation_photons;
+
+    iEvent.getByToken(PUPPINoLeptonsIsolation_charged_hadrons_, PUPPINoLeptonsIsolation_charged_hadrons);
+    iEvent.getByToken(PUPPINoLeptonsIsolation_neutral_hadrons_, PUPPINoLeptonsIsolation_neutral_hadrons);
+    iEvent.getByToken(PUPPINoLeptonsIsolation_photons_, PUPPINoLeptonsIsolation_photons);
+
     std::vector<unsigned int> keys;
     for (View<pat::Electron>::const_iterator it = src->begin(), ed = src->end(); it != ed; ++it) {
         out->push_back(*it);
         pat::Electron & electron = out->back();
+        auto elePtr = src -> ptrAt(it - src->begin());
+        double PUPPI_Isolation = (*PUPPIIsolation_charged_hadrons)[elePtr] + (*PUPPIIsolation_neutral_hadrons)[elePtr] + (*PUPPIIsolation_photons)[elePtr];
+        double PUPPINoLeptons_Isolation = (*PUPPINoLeptonsIsolation_charged_hadrons)[elePtr] + (*PUPPINoLeptonsIsolation_neutral_hadrons)[elePtr] + (*PUPPINoLeptonsIsolation_photons)[elePtr];
+        electron.addUserFloat("PUPPI_Isolation", PUPPI_Isolation);
+        electron.addUserFloat("PUPPINoLeptons_Isolation", PUPPINoLeptons_Isolation);
```

Comment:
the idea for recent updates was that all changes will be removed from the ElectronSlimmer